lib/loader/index.js

```diff
@@ -233,30 +263,34 @@ function postInstantiate(extendedExports, instance) {
-  extendedExports.__getUint8Array = getTypedArray.bind(null, Uint8Array, 0);
-  extendedExports.__getUint8ArrayView = getTypedArrayView.bind(null, Uint8Array, 0);
-  extendedExports.__getUint8ClampedArray = getTypedArray.bind(null, Uint8ClampedArray, 0);
-  extendedExports.__getUint8ClampedArrayView = getTypedArrayView.bind(null, Uint8ClampedArray, 0);
-  extendedExports.__getInt16Array = getTypedArray.bind(null, Int16Array, 1);
-  extendedExports.__getInt16ArrayView = getTypedArrayView.bind(null, Int16Array, 1);
-  extendedExports.__getUint16Array = getTypedArray.bind(null, Uint16Array, 1);
-  extendedExports.__getUint16ArrayView = getTypedArrayView.bind(null, Uint16Array, 1);
-  extendedExports.__getInt32Array = getTypedArray.bind(null, Int32Array, 2);
-  extendedExports.__getInt32ArrayView = getTypedArrayView.bind(null, Int32Array, 2);
-  extendedExports.__getUint32Array = getTypedArray.bind(null, Uint32Array, 2);
-  extendedExports.__getUint32ArrayView = getTypedArrayView.bind(null, Uint32Array, 2);
+  /** Get a typed array from a StaticArray in the module's memory. */
+  function getStaticArray(Type, alignLog2, ptr) {
+    return new Type(getStaticArrayView(Type, alignLog2, ptr));
+  }
+
+  /** Gets a live view on a StaticArray's values in the module's memory. */
+  function getStaticArrayView(Type, alignLog2, ptr) {
+    const buffer = memory.buffer;
+    const U32 = new Uint32Array(buffer);
+    return new Type(buffer, ptr, U32[ptr + SIZE_OFFSET >>> 2] >>> alignLog2);
+  }
+
+  const ctors = [Int8Array, Uint8Array, Uint8ClampedArray,
+                 Int16Array, Uint16Array, Int32Array,
+                 Uint32Array, Float32Array, Float64Array];
+
   if (BIGINT) {
-    extendedExports.__getInt64Array = getTypedArray.bind(null, BigInt64Array, 3);
-    extendedExports.__getInt64ArrayView = getTypedArrayView.bind(null, BigInt64Array, 3);
-    extendedExports.__getUint64Array = getTypedArray.bind(null, BigUint64Array, 3);
-    extendedExports.__getUint64ArrayView = getTypedArrayView.bind(null, BigUint64Array, 3);
+    ctors.push(BigInt64Array, BigUint64Array);
   }
-  extendedExports.__getFloat32Array = getTypedArray.bind(null, Float32Array, 2);
-  extendedExports.__getFloat32ArrayView = getTypedArrayView.bind(null, Float32Array, 2);
-  extendedExports.__getFloat64Array = getTypedArray.bind(null, Float64Array, 3);
-  extendedExports.__getFloat64ArrayView = getTypedArrayView.bind(null, Float64Array, 3);
+
+  ctors.forEach(ctor => {
+    let name = "__get" + ctor.name.slice(0, -5);
+    let align = 31 - Math.clz32(ctor.BYTES_PER_ELEMENT);
+    extendedExports[name + "ArrayView"] = getTypedArrayView.bind(null, ctor, align);
+    extendedExports[name + "Array"] = getTypedArray.bind(null, ctor, align);
+    extendedExports[name + "StaticArrayView"] = getStaticArrayView.bind(null, ctor, align);
+    extendedExports[name + "StaticArray"] = getStaticArray.bind(null, ctor, align);
```

Comment:
Wdyt of naming the static variants `__getStaticInt8ArrayView` etc? Feels a bit more ergonomic to me and also makes this setup code a little simpler (no need to slice name).

Reply:
It's a `StaticArray`, and putting those two words together seems important to me to indicate the purpose of the function. But this is *your* project.

Reply:
Also wondering: Do the `BigInt64Array` ctors now give us `__getBigInt64Array` etc. functions? I think these used to be `__getInt64Array` etc.

Reply:
I'm happy to change it to whatever feels better.

Reply:
Got me thinking: Should we actually handle the static arrays like typed arrays here in the first place? Since there are neither `Int8StaticArray`s nor `StaticInt8Array`s as of my initial suggestion, this might be confusing anyway. Wdyt of designing the API analogous to generic `Array`s instead, `__getStaticArray`?

Reply:
My thoughts:

- Getting specific methods for returning the typed arrays makes things very fast, (even faster than ArrayBufferView functions)
- Naming them `__getStatic*(View?)` is perfectly fine and I can fix that.

I propose renaming the functions as you describe and leaving the api AS-IS.

Reply:
The reason I feel this way is because we can already use the `__getArrayView()` function to obtain the same result without having to worry about the type of array.

Reply:
Still feels like feature creep to me. Perhaps there is a way to refactor the API in general, to less functions, that are still fast?